tests/conftest.py

```diff
@@ -1,42 +1,83 @@
 # SPDX-License-Identifier: MIT
 
+import os
+import os.path
+import shutil
+import stat
+import tempfile
+
 import pytest
 
 
-DUMMY_PYPROJECT = '''
-[build-system]
-requires = ['flit_core']
-build-backend = 'flit_core.buildapi'
-'''.strip()
+def pytest_addoption(parser):
+    parser.addoption('--run-integration', action='store_true', help='run the integration tests')
```

Comment:
I find a better approach is to create an integration folder. That way users can do integration only via ``-k integration``. Then we just need to skip by default all tests in that folder, but enable them if user passed a ```--integration`` flag.

Reply:
`pytest ... tests/test_integration.py` would also work. All these are possible options, I have a slight preference for the current approach since everything is handled the same way. Is there any benefit of handling it as you propose?

Reply:
Reduces the size of API needed to be maintained within ``conftest.py``, and keeps the API more towards what pytest offers, rather than adding new flags/behaviours by a bit.

Reply:
Alright, that makes sense.

Reply:
I didn't change this as I think the maintenance burden here is pretty low and the options do provide a nicer UX for contributors. The tradeoff seems pretty reasonable to me, but if you disagree I can change, let me know :grinning: